scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -582,49 +608,76 @@ def uri(self) -> str:
         # do not exist then fail gracefully, return the URI "as-is", and hope for the best.
         try:
             blob_service_client = BlobServiceClient(
-                account_url=self._account_uri,
+                account_url=account_uri,
                 credential=AzureCliCredential(
                     process_timeout=AzureBlobstoreAssetPath.AZURE_CLI_PROCESS_LOGIN_TIMEOUT
                 )
             )
-            container_client = blob_service_client.get_container_client(container=self._container_name)
+            container_client = blob_service_client.get_container_client(container=container_name)
 
             # If the container allows for anonymous access then we can return the URI "as-is"
             if container_client.get_container_properties().public_access is not None:
-                return self._uri
+                return ""
 
             # Our final approach is to generate a SAS token for the container and append
             # it to the URI
-            start_time = datetime.datetime.now(datetime.timezone.utc)
-            expiry_time = start_time + AzureBlobstoreAssetPath.SAS_EXPIRATION_TIME_DELTA
-
             key = blob_service_client.get_user_delegation_key(start_time, expiry_time)
 
             sas_token = "?" + generate_container_sas(
-                account_name=self._storage_name,
-                container_name=self._container_name,
+                account_name=storage_name,
+                container_name=container_name,
                 user_delegation_key=key,
                 permission=ContainerSasPermissions(read=True, list=True),
                 expiry=expiry_time,
                 start=start_time
             )
 
-            self._uri += sas_token
-            return self._uri
+            return sas_token
 
         except Exception:

```

Comment:
Is the way to log an exception here (ex. in cause of auth error) is through a print statement? It seems like logger can't be used because it causes a circular import

Reply:
You could print to stderr. I've run into the circular import issue too. :(